master/pkg/tasks/task_command.go

```diff
@@ -114,3 +114,38 @@ func (s GenericCommandSpec) ToTaskSpec(keys *ssh.PrivateAndPublicKeys) TaskSpec
 	res.TaskType = s.TaskType
 	return res
 }
+
+// TrialSpec has matching `ProxyPorts` and `MakeEnvPorts` methods. Long-term, we should
+// - unify TrialSpec and GenericCommandSpec
+// - move CommandConfig to expconf
+
+// MakeEnvPorts fills in `Environment.Ports` i.e. exposed ports for container config.
+func (s *GenericCommandSpec) MakeEnvPorts() {
+	if s.Config.Environment.Ports != nil {
+		panic("CommandSpec Environment.Ports are only supposed to be generated once.")
+	}
+
+	ppc := s.ProxyPorts()
+	s.Config.Environment.Ports = map[string]int{}
```

Comment:
why do we overwrite the entire thing, as opposed to just adding `s.Base.ExtraProxyPorts`?